Light touches for catalog vocabulary

ceblanton opened the pull request:
@aradhakrishnanGFDL ran the scripts with sample output from the current catalog builder, and found it almost worked, and needed one small change-- adding `chunk_freq` because it is one of the `aggregate_columns` set in the catalog schema:

https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/main/cats/gfdl_template.json

Also, `frequency` values should eventually match those in this CV location. `monthly` is the Bronx DRS pattern, `P1M` is the Canopy DRS pattern, and `mon` is the agreed standard based on community use:

https://github.com/NOAA-GFDL/CMIP6_CVs/blob/main/CMIP6_frequency.json

aradhakrishnanGFDL committed:
mods to make it work with a new exp catalog. Some hardwires need to be adjusted to make the key retrieval a bit dynamic, but works now.
catlaog builder needs to rm source_id from aggregation

aradhakrishnanGFDL committed:
added a test script

aradhakrishnanGFDL commented:
thanks @ceblanton 

Additional info copied from https://github.com/aradhakrishnanGFDL/CatalogBuilder/issues/120 for reference


figure generated : /nbhome/a1r/analysis-scripts/pngs/cloud-fraction.png

script used: https://github.com/aradhakrishnanGFDL/analysis-scripts/blob/prototype1-a1r/raytest.py

changes made are in my fork
and its only for one suite

https://github.com/aradhakrishnanGFDL/analysis-scripts/tree/prototype1-a1r/freanalysis_clouds

to support this, we need to remove source_id from the aggregation columns (in catalog builder schema). MDTF uses it though. so let's discuss.. @ceblanton

ceblanton commented:
Do we want to dump `source_id` as an aggregate column? I thought we're OK with "na" as a default value so that source_id could still be used for multi-model analysis, say.

aradhakrishnanGFDL committed:
Update __init__.py

added source_if back to key

schema/json 
https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/main/cats/gfdl_template.json

aradhakrishnanGFDL commented:
source_id added back in. I haven't tested it once I added source_id. But why would it not work ?:) 

reference schema here https://github.com/aradhakrishnanGFDL/CatalogBuilder/blob/main/cats/gfdl_template.json
